src/libstd/os/linux/fs.rs

```diff
@@ -24,9 +24,25 @@ pub trait MetadataExt {
     /// Gain a reference to the underlying `stat` structure which contains
     /// the raw information returned by the OS.
     ///
-    /// The contents of the returned `stat` are **not** consistent across
+    /// The contents of the returned [`stat`] are **not** consistent across
     /// Unix platforms. The `os::unix::fs::MetadataExt` trait contains the
     /// cross-Unix abstractions contained within the raw stat.
+    ///
+    /// [`stat`]: ../../../../std/os/linux/raw/struct.stat.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::fs;
+    /// use std::os::linux::fs::MetadataExt;
+    ///
+    /// # use std::io;
+    /// # fn f() -> io::Result<()> {
+    /// let meta = fs::metadata("some_file")?;
+    /// let stat = meta.as_raw_stat();
```

Comment:
Is it worth adding an example to this method, since it's deprecated?

Reply:
It's always worth it. As long as it's there, it has to be documented.

Reply:
Fair point. I was thinking in terms of "we're suggesting people use other methods, so why do we provide incentive to look at this one?"

Also now that i'm thinking about it, i'm concerned that this may not actually pass tests, since every doctest in std has `deny(warnings)` applied to it. If this passes travis, i'll let it slide, but that's at least my reasoning.